src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -1317,13 +1317,62 @@ Namespace Microsoft.CodeAnalysis.Semantics
+
+            Dim eventInvocation = TryCast(boundRaiseEventStatement.EventInvocation, BoundCall)
+
+            ' Return an invalid statement for invalid raise event statement
+            If eventInvocation Is Nothing OrElse eventInvocation.ReceiverOpt Is Nothing
+                Debug.Assert(boundRaiseEventStatement.HasErrors)
+                Dim children As Lazy(Of ImmutableArray(Of IOperation)) = New Lazy(Of ImmutableArray(Of IOperation))(Function() ImmutableArray.Create(Of IOperation)(Create(boundRaiseEventStatement.EventInvocation)))
+                return New LazyInvalidStatement(children, _semanticModel, syntax, type, constantValue, isImplicit)
+            End If 
+            
+            Dim receiver = eventInvocation.ReceiverOpt
+            Dim eventSymbol = boundRaiseEventStatement.EventSymbol
+            Dim eventReferenceSyntax = receiver.Syntax
+            Dim eventReferenceType As ITypeSymbol = eventSymbol.Type
+            Dim eventReferenceConstantValue As [Optional](Of Object) = ConvertToOptional(receiver.ConstantValueOpt)
+            ' EventReference in a raise event statement is never implicit. However, the way it is implemented, we don't get
+            ' a "BoundEventAccess" for either field backed event or custom event, and the bound nodes we get are marked as 
+            ' generated by compiler. As a result, we have to explicitly set IsImplicit to false.
+            dim eventReferenceIsImplicit as Boolean = False
+            
+            Dim boundInstance As BoundNode
+            If receiver.Kind = BoundKind.FieldAccess
+                ' For raising a field backed event, we will only get a field access node in bound tree.
+                Dim eventFieldAccess = CType(receiver, BoundFieldAccess)
+                Debug.Assert(eventFieldAccess.FieldSymbol.AssociatedSymbol = eventSymbol)
+
+                boundInstance = eventFieldAccess.ReceiverOpt
+            Else 
+                ' This is for a custom event
+                boundInstance = receiver
+            End If
+
+            Dim eventReferenceInstance As Lazy(Of IOperation) = New Lazy(Of IOperation)(Function() If(eventSymbol.IsShared, Nothing, Create(boundInstance)))
+            
+            Dim EventReference As Lazy(Of IEventReferenceExpression) = New Lazy(Of IEventReferenceExpression)(Function() As IEventReferenceExpression
+                                                                                                                  Return New LazyEventReferenceExpression(eventSymbol, 
+                                                                                                                                                          eventReferenceInstance, 
+                                                                                                                                                          eventSymbol, 
```

Comment:
>eventSymbol [](start = 154, length = 11)

Why are we passing eventSymbol twice? Do we have the same duplication of information that we used to have with method reference?

Reply:
Please open an issue to follow up.

---
In reply to: [143185433](https://github.com/dotnet/roslyn/pull/22243#discussion_r143185433) [](ancestors = 143185433)

Reply:
Yes, we have both `MemberReferenceExpression.Member` and `MemberReferenceExpression.Event` referring to the same event symbol.

Reply:
And it seems this is still an issue for `BaseMethodReferenceExpression` as well. Do we already have an issue tracking that (I couldn't find one)? @dotnet/analyzer-ioperation

Reply:
> And it seems this is still an issue for BaseMethodReferenceExpression as well.

I believe @333fred is fixing method references in https://github.com/dotnet/roslyn/pull/22056.

Reply:
Yes, that's fixed in my PR.